Marlin/Configuration.h

```diff
@@ -580,6 +587,43 @@
 #define THERMAL_PROTECTION_BED     // Enable thermal protection for the heated bed
 #define THERMAL_PROTECTION_CHAMBER // Enable thermal protection for the heated chamber
 
+//===========================================================================
+//====================== Thermal First Layer Compensation ===================
+//===========================================================================
+
+/**
+ * Thermal first layer compensation for mesh bed leveling. Compensation z-offset
+ * values will be added to the probing measurements when running G29 command.
+ * Calibrate bed and probe using G76. Modify values manually using M871.
+ * Find a more detailed explaination of the calibration process in file
+ * src/gcode/calibrate/G76_M871.cpp
+ */
+#if TEMP_SENSOR_BED && TEMP_SENSOR_PROBE
+  // Enable thermal first layer compensation using bed and probe temperatures
+  #define USE_TEMP_COMPENSATION
+
+  // Max temperature that can be reached by heated bed.
+  // This is required only for the calibration process.
+  #define TEMP_COMPENSATION_MAX_BED_TEMP 110
+  
```

Comment:
Could this be derived from `MAX_BED_TEMP`?

Reply:
Yes, assuming users set MAX_BED_TEMP correctly in firmware, i.e. the bed can actually reach that temperature. Otherwise they will run into temperature errors while calibrating.